README.md

```diff
@@ -195,6 +198,70 @@ We have a small collection of algorithms, one for every day of the month. Scroll
   ```
 - **Resources**
   -[Matrix: Determinant](https://www.mathsisfun.com/algebra/matrix-determinant.html)
+
+### December 11 - Printing matrix in spiral form
+- **Problem**
+  - Given a Matrix of size `M * N`, output its elements in spiral form.
+  - To better understand the problem statement, consider the following matrix
+  ```
+  01     02     03
+  04     05     06
+  07     08     09
+   ```
+  - The Spiral form of a matrix is as below.
+    ```
+    01 --> 02 --> 03 --> 06 --> 09 --> 08 --> 07 --> 04 --> 05
+    ```
+  - Move Right, Move Down, Move Left, Move Up till you cover all elements of matrix.
+- **Uses**
+  - Matrices are used much more in daily life than people would have thought. In fact it is in front of us every day when going to work, at the university and even at home.
+  - [Real Applications of Matrices](https://www.ukessays.com/essays/mathematics/application-of-matrices-in-real-life-problems.php)
+- **Resources**
+  - Matrices are represented using two dimensional arrays in any . So to access a single element in the matrix two “for loops”
+  - [Operations on Matrices](https://www.geeksforgeeks.org/different-operation-matrices/)
+
+
+### December 12 - Reversing a singly linked list
+- **Problem**
+  - Taking a singly linked as the input the output has to be another singly linked list having the reverse of the initial linked list
+- **Example**
+  ```
+  Input:
+    10 -> 20 -> 34 -> 23 -> 889
+  Output:
+    889 -> 23 -> 34 -> 20 -> 10
+  ```
+- **Uses**
+  - There are pretty good real examples to show the usage and importance of linked-list.
+  - Consider the history section of web browsers, where it creates a linked list of web-pages visited, so that when you check history (traversal of a list) or press back button, the previous node's data is fetched.
+  - One common sighted example is low level memory management (i.e. the heap as managed by `malloc` in C or new in Java, etc) is often implemented as a linked list, with each node representing a used or available (free) block of memory. These blocks may be of any size, change size (combine and split), be freed or assigned in any order, and reordered. A linked list means you can keep track of all of these "nodes" and manipulate them fairly easily.
+  - Also, Hash-tables that use chaining to resolve hash collisions typically have one linked list per bucket for the elements in that bucket.
+- **Resources:**
+  - [This is how linked lists work](https://www.youtube.com/watch?v=NobHlGUjV3g)
+  - `Clue:` This problem requires the use of another data structure
+
+### December 13 -  Lexicographical arrangement
+- **Problem**
+  - In this question the input is a string with a given number of characters. The problem here is to arrange all the possible permutations of this string in the alphabetical order. The next step is to find out the position at which this string occurs in this order.
```

Comment:
Change it to "Given an input string, find all possible permutations of this string. Arrange these permutations in alphabetical order and find the position at which the input string occurs.